recipes/openssl/1.x.x/conanfile.py

```diff
@@ -413,6 +414,8 @@ def _configure_args(self):
         if self._full_version >= "1.1.0":
             args.append("--debug" if self.settings.build_type == "Debug" else "--release")
 
+        if str(self.settings.os) == "tvOS":
```

Comment:
```suggestion
        if self.settings.os == "tvOS":
```

Reply:
If this is changed, then the rest of the `if self.settings.os ==` statements that follow should also be changed. I didn't notice it before, but the usage of this is horribly inconsistent though the entire recipe. Is there a standardization somewhere that says one should be used over the other? I typically don't do "standardization" type changes for commits for things like this, but I'm happy to do it if you'd prefer it.

Reply:
By dropping the `str`, `conan` is able to check whether `tvOS` is a valid os.
That way typos can be catched.
If the code is equivalent, I'ld say 'why not?'

Reply:
Yeah, it doesn't matter to me one way or the other - I suck at Python & honestly don't know the difference.

Just because I'm curious and would like to know - how is it that Conan can check if it's a valid OS by doing `self.settings.os` rather than `str(self.settings.os)`? If `self.settings.os` is a string, then using `str()` really just turns an existing string into a string, right?

Maybe a better way to put it is - how is it that dropping `str()` makes it so Conan can check if `tvOS` is a valid OS?

Reply:
When you compare against `self.setyinf.os`, you're comparing against a SettingsItem object.
This class checks whether strings to which it is compared against would also be a valid setting (ie it is present in your settings.yml file).
By doing `str`, you compare against a string without an additional check.